test/client.test.js

```diff
@@ -158,34 +174,66 @@ describe("ServiceClient", () => {
     client.request().catch(err => {
       assert(err instanceof ServiceClient.Error);
       assert.equal(err.type, ServiceClient.BODY_PARSE_FAILED);
+      assert(err instanceof BodyParseError);
       assert.deepStrictEqual(err.response, response);
       done();
     });
   });
 
   it("should give a custom error object when request fails", () => {
     const client = new ServiceClient(clientOptions);
-    const requestError = new Error("foobar");
+    const requestError = new RequestError("foobar");
     requestStub.returns(Promise.reject(requestError));
     return client.request().catch(err => {
       assert(err instanceof ServiceClient.Error);
       assert.equal(err.type, ServiceClient.REQUEST_FAILED);
+      assert(err instanceof RequestFailedError);
+    });
+  });
+
+  it("should give a custom error when request timeouts", () => {
+    const client = new ServiceClient(clientOptions);
+    requestStub.rejects(new ConnectionTimeoutError("foobar"));
+    return client.request().catch(err => {
```

Comment:
This pattern of testing errors doesn't fail when the promise doesn't reject. How about using `done()` callback, or something like:

```js
const expectRejection = () => {
  assert.fail('Expected a promise to reject');
};

// ...
return client.request().then(expectRejection, err => {
  assert(...);
});
```

Reply:
@shuhei Good catch, thank you. There were actually a couple of tests that weren't checking anything. I went with your second suggestion as it fails faster and provides better a error message. Pushed a new commit.